src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,186 @@
+                    success = true;
+
+                    return result;
+                }
+                catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.ServiceUnavailable || e.Status == HttpStatusRequestThrottled)
+                {
+                    latestException = e;
+                    continue;
+                }
+                catch (AggregateException ex) when (ex.InnerExceptions?.All(e => e is RequestFailedException) == true)
+                {
+                    var lastException = ex.InnerExceptions.LastOrDefault() as RequestFailedException;
+
+                    if (lastException?.Status == (int)HttpStatusCode.ServiceUnavailable || lastException?.Status == HttpStatusRequestThrottled)
+                    {
+                        latestException = lastException;
+                        continue;
+                    }
+
+                    throw;
+                }
+                catch (OperationCanceledException)
+                {
+                    operationCanceled = true;
+                    throw;
+                }
+                finally
+                {
+                    if (i == 0 && !operationCanceled)
+                    {
+                        UpdatePrimaryConfigStoreStatus(success);
+                    }
+                }
+            }
+
+            throw latestException;
+        }
+
+        private T ExecuteWithFailOverPolicy<T>(IEnumerable<Func<T>> delegates, CancellationToken cancellationToken = default)
+        {
+            Exception latestException = null;
+
+            for (var i = ShouldTryPrimaryConfigStore() ? 0 : 1; i < clients.Count(); i++)
+            {
+                var success = false;
+                var operationCanceled = false;
+
+                try
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    var result = delegates.ElementAt(i)();
+                    success = true;
+
+                    return result;
+                }
+                catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.ServiceUnavailable || e.Status == HttpStatusRequestThrottled)
+                {
+                    latestException = e;
+                    continue;
+                }
+                catch (AggregateException ex) when (ex.InnerExceptions?.All(e => e is RequestFailedException) == true)
+                {
+                    var lastException = ex.InnerExceptions.LastOrDefault() as RequestFailedException;
+
+                    if (lastException?.Status == (int)HttpStatusCode.ServiceUnavailable ||
+                        lastException?.Status == HttpStatusRequestThrottled)
+                    {
+                        latestException = lastException;
+                        continue;
+                    }
+
+                    throw;
+                }
+                catch (OperationCanceledException)
```

Comment:
If a region is down, the client won't get any responses. We will end up in this case. If a user provides a replica, shouldn't we try other replicas. I would argue this is probably the most important scenario for using a fail-over replica.

How long will it take to fail here in case the region is down? I am wondering if we should try even more aggressively for config providers who have replica configured.